OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -46,6 +47,7 @@ public class Move : Activity
 		public Move(Actor self, CPos destination)
 		{
 			mobile = self.Trait<Mobile>();
+			notifyMoving = self.TraitsImplementing<INotifyMoving>().ToArray();
```

Comment:
Tracking this in N places isn't ideal IMO.

Is there any reason why we can't do this inside `Mobile` by overloading the `IsMoving` setter?

```csharp
bool isMoving;
public bool IsMoving
{
	get
	{
		return isMoving;
	}
	set
	{
		if (value && !isMoving)
			foreach (var n in notifyMoving)
				n.StartedMoving(self);
		else if (!value && isMoving)
			foreach (var n in notifyMoving)
				n.StoppedMoving(self);

		isMoving = value;
	}
}
```

This would simplify the diff a lot, and make it a lot harder to accidentally forget to notify the interface consumers.

Reply:
This works for Mobile, but not so well for Aircraft.
Problem is that for aircraft, traits like WithMoveAnimation sometimes need to evaluate both IsMoving and IsMovingVertically from the notification methods. However, with the above approach, the IsMoving notifications are triggered before IsMovingVertically's value is updated, even if their value changes on the same tick. This can lead to wrong results.

So at least for `Aircraft`, I see no other way than taking the 'old' approach for now (but maybe with a dedicated method which also contains an explaining comment).

Reply:
Yeah, I only meant this for `Mobile`. Sorry if that wasn't clear.